### PR TITLE
Bug 1713671: Move extended build tests from ruby:2.x to ruby:2.5

### PR DESCRIPTION
**Pull request by gabemontero**
need to fix ruby based extended build test in order to merge the openshift/builder bug 1713671 via https://github.com/openshift/builder/pull/97

So this is a pick of https://github.com/openshift/origin/pull/23452 which had no BZ assoiciated in the 4.2 timeframe

/assign @adambkaplan 
@openshift/openshift-team-developer-experience  FYI

### CODE REVIEW

**Comment by gabemontero** [comment]
fyi @adambkaplan one of the files in the 4.2 commit does not exist in 4.1

**Comment by gabemontero** [comment]
looks like we need to update the cmd tests in 4.1 as well @adambkaplan 

there are several ruby 2.2 related failures there

I believe with 4.2 we didn't have to worry about them since the cmd tests were removed 

I'll circle back to this in a bit

**Comment by gabemontero** [comment]
cmd tests are now passing 

/retest

**Comment by gabemontero** [comment]
bump clean running tests @adambkaplan

File: test/cmd/newapp.sh [hunk 1]
```diff
@@ -576,42 +568,6 @@ os::cmd::expect_failure_and_text 'oc new-build --name sourcetest --source-image
...
-
-os::cmd::expect_success 'oc new-app php-55-centos7:latest --code https://github.com/openshift/ruby-hello-world.git --name test6'
-os::cmd::expect_success_and_text 'oc get bc test6 --template={{.spec.strategy.sourceStrategy.from.name}}' 'php-55-centos7:latest'
-
-os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world.git --name test7'
-os::cmd::expect_success_and_text 'oc get bc test7 --template={{.spec.strategy.dockerStrategy.from.name}}' 'ruby-22-centos7:latest'
-
-os::cmd::expect_success 'oc new-app php-55-centos7:latest https://github.com/openshift/ruby-hello-world.git --name test8'
-os::cmd::expect_success_and_text 'oc get bc test8 --template={{.spec.strategy.sourceStrategy.from.name}}' 'php-55-centos7:latest'
-os::cmd::expect_success 'oc delete project new-app-syntax'
```
Comment (adambkaplan) [inline test/cmd/newapp.sh]: @gabemontero are these tests valid any more? Can these be upgraded to use non-EOL images?

Comment (gabemontero) [reply test/cmd/newapp.sh]: @adambkaplan perhaps ... but in the context of the cmd tests totally being removed in 4.2 and beyond, and all these tests disappearing, I chose not the invest the additional time to try and sort out alternatives

Comment (gabemontero) [reply test/cmd/newapp.sh]: boils down to how strongly we feel about it from a 4.1.x only context IMO

Comment (gabemontero) [reply test/cmd/newapp.sh]: also, it isn't like an actual cluster, where we have the complete set of samples available ... I did try `openshift/ruby:2.5` in the ruby case and it didn't work

Comment (adambkaplan) [reply test/cmd/newapp.sh]: right - makes sense to drop then.

**Comment by gabemontero** [comment]
looks like I've touched enough files that we need more approvers

/assign @bparees

**Comment by bparees** [comment]
/approve

we'll call it just under the wire on 4.2 code freeze.....
(test code only changes, thus no risk to release)

**Comment by gabemontero** [comment]
On Fri, Sep 20, 2019 at 6:15 PM Ben Parees <notifications@github.com> wrote:

> /approve
>
> we'll call it just under the wire on 4.2 code freeze.....
> (test code only changes, thus no risk to release)
>
Fyi this is a 4.1 pr to enable a openshift/builder cherry pick

**Comment by bparees** [comment]
@gabemontero thanks, I missed that. 4.1 got closed early this week, so this will have to wait for next. You'll make the your and the patch manager's life easier if you retroactively create a 4.2 bz and get it verified.

**Comment by gabemontero** [comment]
On Sat, Sep 21, 2019 at 9:59 AM Ben Parees <notifications@github.com> wrote:

> @gabemontero <https://github.com/gabemontero> thanks, I missed that. 4.1
> got closed early this week, so this will have to wait for next. You'll make
> the your and the patch manager's life easier if you retroactively create a
> 4.2 bz and get it verified.
>
the first commit has a 4.2 bz - cherry picked that one from there @bparees

The second commit is a no-op in 4.2 because the cmd tests have been removed

**Comment by gabemontero** [comment]
OK got to a browser and saw my PR's description:

`So this is a pick of #23452 which had no BZ assoiciated in the 4.2 timeframe`

I'll get a bz created for that.

**Comment by gabemontero** [comment]
OK https://bugzilla.redhat.com/show_bug.cgi?id=1754186 has been retroactively created for the 4.2/master PR which adjusted the extended tests.

FYI given we are past code freeze, the only available release to assign to was 4.2.z @bparees 

I took a guess and made the target 4.2.0 since the test change is already in 4.2/master.  But please advise / correct as needed.

**Comment by bparees** [comment]
@gabemontero can you link the PR that merge the test updates to master, in https://bugzilla.redhat.com/show_bug.cgi?id=1754186 and then move it to MODIFIED?  (once it then goes to ON_QA you can probably just move to verified yourself, rather than rely on QE given the nature of the changes)

**Comment by gabemontero** [comment]
On Sat, Sep 21, 2019 at 6:45 PM Ben Parees <notifications@github.com> wrote:

> @gabemontero <https://github.com/gabemontero> can you link the PR that
> merge the test updates to master, in
> https://bugzilla.redhat.com/show_bug.cgi?id=1754186 and then move it to
> MODIFIED? (once it then goes to ON_QA you can probably just move to
> verified yourself, rather than rely on QE given the nature of the changes)
>
already did this morning @bparess and it was immediately moved to MODIFIED
but I then somehow accidentally moved it back to NEW

re-moving it to MODIFIED now

**Comment by gabemontero** [comment]
Bug went to ON_QA, and I've moved it to VERIFIED

**Comment by adambkaplan** [comment]
**PATCH MANAGER NOTE**

This PR blocks https://github.com/openshift/builder/pull/97 from passing the `e2e-aws-builds` suite, which is required for merge.

**Comment by adambkaplan** [comment]
**PATCH MANAGER NOTE**

Also blocks openshift/builder#94

**Pull request merged**